stl/inc/xstring

```diff
@@ -2290,8 +2298,8 @@ public:
     };
     _Bxty _Bx;
 
-    size_type _Mysize = 0; // current length of string
-    size_type _Myres  = 0; // current storage reserved for string
+    size_type _Mysize = 0; // current length of string (string::size(); <= _Myres)
+    size_type _Myres  = _SMALL_STRING_CAPACITY; // current storage reserved for string (string::capacity())

```

Comment:
I realize that `_Myres` should be initialized to `_SMALL_STRING_CAPACITY`, as `_String_val` begins with SSO mode.
After this change, however, I'm getting `static_assert(test_to_string<63>()); ... constexpr evaluation hit maximum step limit` in [this test](https://github.com/llvm/llvm-project/blob/dc6c8b8d1e357acf6440824afaf6b6547b34aeeb/libcxx/test/std/utilities/template.bitset/bitset.members/to_string.pass.cpp#L118).

Reply:
If this change is valid, then some further changes are needed(e.g. removing redundant assignments like `_Myres = _SMALL_STRING_CAPACITY` in basic_string's constructors)

Reply:
> After this change, however, I'm getting `static_assert(test_to_string<63>()); ... constexpr evaluation hit maximum step limit` in [this test](https://github.com/llvm/llvm-project/blob/dc6c8b8d1e357acf6440824afaf6b6547b34aeeb/libcxx/test/std/utilities/template.bitset/bitset.members/to_string.pass.cpp#L118).

This is because of redundant checking. Do you think it make sense to check in `_Large_string_engaged` which is heavily called?

Reply:
I intended to use `_INTERNAL_CHECK` to express invariant here. I think it will not expand to live code in most cases.
However, yes, I'm uncertain whether this is good practice. As the function is only called in several places, it might be useless and just some noise. And due to early call it's introducing a stricter invariant: `_Myres` should >= `_SMALLSTRING_CAPACITY`, `from the beginning of lifetime`(this looks not too bad for me).

Reply:
I decide to remove this check to pass those clang tests.